receiver/awsfirehosereceiver/metrics_receiver.go

```diff
@@ -69,8 +69,10 @@ func (c *metricsConsumer) Start(_ context.Context, host component.Host) error {
 		// TODO: make cwmetrics an encoding extension
 		c.unmarshaler = cwmetricstream.NewUnmarshaler(c.settings.Logger, c.settings.BuildInfo)
 	case otlpmetricstream.TypeStr:
-		// TODO: make otlp_v1 an encoding extension
-		c.unmarshaler = otlpmetricstream.NewUnmarshaler(c.settings.Logger, c.settings.BuildInfo)
+		var err error
+		if c.unmarshaler, err = otlpmetricstream.NewUnmarshaler(ctx); err != nil {
```

Comment:
Should we create the encoding extension directly here? Then we can use the receiver.Settings for creating the extension.Settings